addon/serializers/github-tree.js

```diff
@@ -0,0 +1,33 @@
+import GithubSerializer from './github';
+
+export default GithubSerializer.extend({
+  normalize(modelClass, resourceHash, prop) {
+    let normalizedHash = {
+      id: resourceHash.sha,
+      sha: resourceHash.sha,
+      url: resourceHash.url,
+      files: resourceHash.tree
+        .filter(item => item.type === 'blob')
+        .reduce((files, blob) => {
+          files[blob.path] = blob.sha;
+          return files;
+        }, {}),
+      directories: resourceHash.tree
+        .filter(item => item.type === 'tree')
+        .reduce((files, tree) => {
+          files[tree.path] = tree.sha;
+          return files;
+        }, {}),
+      truncated: resourceHash.truncated,
+      links: {
+        blobs: resourceHash.tree
+          .filter(item => item.type === 'blob')
+          .map(blob => blob.url),
+        trees: resourceHash.tree
+          .filter(item => item.type === 'tree')
```

Comment:
Since you're having to do both of these filters twice, maybe it makes sense to alias them at the top of the method and then perform the reduce/map calls on the already filtered arrays.